src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -1439,6 +1466,87 @@ public IndexMergeResults ProposeIndexMergeSuggestions()
 
             return indexMerger.ProposeIndexMergeSuggestions();
         }
+
+        public class IndexBatchScope
+        {
+            private readonly IndexStore _store;
+
+            private PutIndexesCommand _command;
+
+            public IndexBatchScope(IndexStore store)
+            {
+                _store = store;
+            }
+
+            public void AddIndex(IndexDefinitionBase definition)
+            {
+                if (_command == null)
+                    _command = new PutIndexesCommand(_store._documentDatabase.Name);
+
+                if (definition == null)
+                    throw new ArgumentNullException(nameof(definition));
+
+                if (definition is MapIndexDefinition indexDefinition)
+                {
+                    AddIndex(indexDefinition.IndexDefinition);
+                    return;
+                }
+
+                _store.ValidateAutoIndex(definition);
+
+                var autoDefinition = (AutoIndexDefinitionBase)definition;
+                var indexType = PutAutoIndexCommand.GetAutoIndexType(autoDefinition);
+
+                _command.Auto.Add(PutAutoIndexCommand.GetAutoIndexDefinition(autoDefinition, indexType));
+            }
+
+            public void AddIndex(IndexDefinition definition)
+            {
+                if (_command == null)
+                    _command = new PutIndexesCommand(_store._documentDatabase.Name);
+
+                _store.ValidateStaticIndex(definition);
+
+                _command.Static.Add(definition);
+            }
+
+            public async Task SaveAsync()
+            {
+                if (_command == null || _command.Static.Count == 0 && _command.Auto.Count == 0)
+                    return;
+
+                try
+                {
+                    var (index, _) = await _store._serverStore.SendToLeaderAsync(_command);
+
+                    var indexCount = _command.Static.Count + _command.Auto.Count;
+                    var operationTimeout = _store._serverStore.Engine.OperationTimeout;
+                    var timeout = TimeSpan.FromSeconds((indexCount / 50.0) * operationTimeout.TotalSeconds);
+                    if (operationTimeout > timeout)
+                        timeout = operationTimeout;
+
+                    await _store._documentDatabase.RachisLogIndexNotifications.WaitForIndexNotification(index, timeout);
```

Comment:
Note that this will ensure the execution only on this specific node and not on the entire database group.

Reply:
That is enough for us here IMO. Same as in `CreateIndex`.